File: 169-MajorityElement/169-MajorityElement.js
// Last updated: 10/03/2026, 13:45:22
var majorityElement = function(nums) {

    let count = 0;
    let candidate = null;

    for (let num of nums) {

        if (count === 0) {
            candidate = num;
        }

        if (num === candidate) {
            count++;
        } else {
            count--;
        }
    }

    return candidate;
};

File: 169-MajorityElement/169-MajorityElement.cpp
// Last updated: 07/04/2026, 10:35:48
1class Solution {
2public:
3    void moveZeroes(vector<int>& nums) {
4        int j = 0;
5
6        for (int i = 0; i < nums.size(); i++) {
7            if (nums[i] != 0) {
8                nums[j++] = nums[i];
9            }
10        }
11
12        while (j < nums.size()) {
13            nums[j++] = 0;
14        }
15    }
16};